packages/plugin-ext/doc/how-to-add-new-custom-plugin-api.md

```diff
@@ -5,18 +5,24 @@ That will require API that goes beyond what's in the VS Code Extension API and t
 You can do that by implementing a Theia extension that creates and exposes an API object within the plugin host.
 The API object can be imported by your plugins and exposes one or more API namespaces.
 
-Depending on the plugin host we can either provide a frontend or backend plugin API:
+Depending on the plugin host we can either provide a frontend or backend plugin API, or an API for headless plugins that extend backend services that are not tied to any frontend connection:
 
 - In the backend plugin host that runs in the Node environment in a separate process, we adapt the module loading to return a custom API object instead of loading a module with a particular name.
+There is a distinct plugin host for each connected Theia frontend.
 - In the frontend plugin host that runs in the browser environment via a web worker, we import the API scripts and put it in the global context.
+There is a distinct plugin host for each connected Theia frontend.
+- In the headless plugin host that also runs in the Node environment in a separate process, we similarly adapt the module loading mechanism.
+There is only one plugin host for headless plugins, if there are any headless plugins deployed.
```

Comment:
Does that also imply that headless plugins cannot be installed during runtime but only during startup/deployment time?

Reply:
No, if they are deployed later, then the plugin-host will start at that time. Will clarify.